service/matching/tasklist/task_list_manager.go

```diff
@@ -258,6 +264,8 @@ func (c *taskListManagerImpl) AddTask(ctx context.Context, params AddTaskParams)
 	if params.ForwardedFrom == "" {
 		// request sent by history service
 		c.liveness.MarkAlive()
+		c.qpsReporter.ReportCounter(1)
+		c.scope.UpdateGauge(metrics.EstimatedAddTaskQPSGauge, c.qpsReporter.QPS())
```

Comment:
could we pass `c.scope` to the `qpsReporter` so it can handle this `UpdateGauge` operation as well?

Reply:
I'll not do that for now. This is a new component and I want to keep its function as minimal as possible so that we could iterate on this easily.